File: 킬링캠프/2주 차 - 리스트, 큐, 스택, 그래프, DFS, BFS/challenge/두 큐 합 같게 만들기/Java/two pointer/Solution.java
class Solution {
    public int solution(int[] queue1, int[] queue2) {
        int n = queue1.length;
        int[] queue = new int[queue1.length + queue2.length];
        System.arraycopy(queue1, 0, queue, 0, queue1.length);
        System.arraycopy(queue2, 0, queue, n, queue2.length);

        long target = (calculateSum(queue1) + calculateSum(queue2)) / 2;
        long queue1Sum = calculateSum(queue1);

        int left = 0;
        int right = queue1.length;
        for (int answer = 0; answer < 3 * n; answer++) {
            if (queue1Sum == target) {
                return answer;
            }

            if (queue1Sum > target) {
                queue1Sum -= queue[left];
                left = (left + 1) % queue.length;
            } else {
                queue1Sum += queue[right];
                right = (right + 1) % queue.length;
            }
        }

        return -1;
    }

    private long calculateSum(int[] queue) {
        long result = 0;
        for (int value : queue) {
            result += value;
        }

        return result;
    }
}

File: 킬링캠프/2주 차 - 리스트, 큐, 스택, 그래프, DFS, BFS/challenge/두 큐 합 같게 만들기/two pointer/solution.py
def solution(queue1, queue2):
    queue_size = len(queue1)
    total_queue = queue1 + queue2
    total_sum = sum(total_queue)
    queue1_sum = sum(queue1)
    queue2_sum = sum(queue2)
    left = 0
    right = len(queue1)

    cnt = 0
    while cnt < queue_size * 4:
        if (queue1_sum == queue2_sum):
            return cnt
        elif queue1_sum > queue2_sum:
            queue1_sum -= total_queue[left]
            queue2_sum = total_sum - queue1_sum
            left = (left + 1) % len(total_queue)
        else:
            queue1_sum += total_queue[right]
            queue2_sum = total_sum - queue1_sum
            right = (right + 1) % len(total_queue)
        
        cnt += 1
    
    return -1
